graph-db/src/utils/bootstrapGremlinServer.js

```diff
@@ -0,0 +1,54 @@
+const { initializeGremlinConnection } = require('./gremlin/initializeGremlinConnection')
+const { indexCmrCollection } = require('./indexing/indexCmrCollection')
+const { fetchPageFromCMR } = require('./cmr/fetchPageFromCmr')
+const { clearScrollSession } = require('./cmr/clearScrollSession')
+const { getEchoToken } = require('./cmr/getEchoToken')
+
+/**
+ * Harvests collections from CMR and loads them into graph db
+ * @returns {Array} Errors from indexing into graph db
+ */
+exports.bootstrapGremilinServer = async () => {
+  const echoToken = await getEchoToken()
+  const gremlin = initializeGremlinConnection()
+  const response = await fetchPageFromCMR(null, echoToken)
+  const results = (await response.json()).items
+  const scrollId = response.headers.get('CMR-Scroll-Id')
+  const partitionedSearchResults = []
+  const indexingStatuses = []
+  let continueScroll = true
+
+  partitionedSearchResults.push(results)
+  while (continueScroll) {
+    // eslint-disable-next-line no-await-in-loop
+    const scrolledResults = await fetchPageFromCMR(scrollId, echoToken)
+      .then((scrollResponse) => scrollResponse.json())
+      .then((json) => {
+        if (json.errors) {
+          throw new Error(`The following errors ocurred: ${json.errors}`)
+        } else {
+          return json.items
+        }
+      })
+      .catch((error) => {
+        console.warn(`Could not complete request due to error: ${error}`)
+        return null
+      })
+
+    partitionedSearchResults.push(scrolledResults)
+
+    if (!scrolledResults || scrolledResults.length < process.env.PAGE_SIZE) {
+      continueScroll = false
+    }
+
+    scrolledResults.forEach(async (result) => {
+      const status = await indexCmrCollection(result, gremlin)
```

Comment:
I would wrap this in a try/catch because as it stands any failure to index will fail the remaining pages of the indexing. If you wrap it in a try/catch if one page fails to index the job will continue attempting to index the remaining pages. If you want to prevent further indexing though, you could ignore this.